### DISCUSSION TITLE
Why does app_local_data_dir() return a Result?

### DISCUSSION THREAD

**Discussion by danielniccoli** [tauri-apps/tauri#10098] [Q&A]
I'm new to Rust and Tauri. I was wondering why `app_local_data_dir()` returns a Result. Shouldn't I as a developer expect that Tauri manages these folders for me and ensures that they exist, when I need to access them? Consequently, if they do not exist, am I supposed to create them manually? (Which I do not think is a good idea.) Hence, should Tauri not fix the problem or panic, instead of asking me what to do about it?

**Comment by FabianLars** [comment]
> Shouldn't I as a developer expect that Tauri manages these folders for me and ensures that they exist, when I need to access them?

Nope. Tauri never creates folders it doesn't need itself.

> Consequently, if they do not exist, am I supposed to create them manually?

Yes.

---

This is not why it returns an error btw, some of the path apis are platform dependend (none of those that are currently exposed in v2 though), or the the inner [dirs](https://docs.rs/dirs/latest/dirs/) library which Tauri uses simply fails to get the base paths for us.
